tests/cpp/test_host_ir_integration.cpp

```diff
@@ -244,8 +244,23 @@ TEST_F(HostIrIntegrationTest, InsertDeallocations) {
                     "from a caching allocator.";
   }
 
-  // At any given time a max of four 32x32 tensors are allocated
-  EXPECT_EQ(max_memory_allocated, sizeof(double) * (32 * 32) * 4);
+  /*
+    At any given time a max of four 32x32 tensors are allocated:
+
+    Here is the flow in the container:
+    1) Input T0_g_double -> 1 tensor allocated
+    2) Allocate T2_g_double -> 2 tensors
```

Comment:
Is this the same as `t2`? I'm surprised `t2` is allocated because it's supposed to be an internal tensor in the second segment.

Reply:
The numbering was off, I updated it to reflect the tv naming above.